http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcRequestAuthenticator.java

```diff
@@ -416,4 +484,112 @@ private static boolean hasScope(String scopeParam, String targetScope) {
+        }
+        if (!deployment.getRequestSignatureAlgorithm().equals(NONE) && deployment.getClientKeyStoreFile() == null){
+            throw log.invalidKeyStoreConfiguration();
+        } else {
+            return loadKeyPairFromKeyStore(deployment.getClientKeyStoreFile(),
+                    deployment.getClientKeyStorePassword(), deployment.getClientKeyPassword(),
+                    deployment.getClientKeyAlias(), deployment.getClientKeystoreType());
+        }
+    }
+
+    public JsonWebSignature signRequest(JwtClaims jwtClaims) throws Exception {
+        JsonWebSignature jsonWebSignature = new JsonWebSignature();
+        jsonWebSignature.setPayload(jwtClaims.toJson());
+
+        if (deployment.getRequestSignatureAlgorithm().contains(NONE)) { //unsigned
+            jsonWebSignature.setAlgorithmConstraints(AlgorithmConstraints.NO_CONSTRAINTS);
+            jsonWebSignature.setAlgorithmHeaderValue(NONE);
+        } else if (deployment.getRequestSignatureAlgorithm().contains(HS_256)
+                || deployment.getRequestSignatureAlgorithm().contains(HS_384)
+                || deployment.getRequestSignatureAlgorithm().contains(HS_512)) { //signed with symmetric key
+            jsonWebSignature.setAlgorithmHeaderValue(deployment.getRequestSignatureAlgorithm());
+            Key key = new HmacKey(deployment.getResourceCredentials().get("secret").toString().getBytes(StandardCharsets.UTF_8));   //the client secret is a shared secret between the server and the client
+            jsonWebSignature.setDoKeyValidation(false);     //skips validation so that size of the secret does not matter
+            jsonWebSignature.setKey(key);
+        } else { //signed with asymmetric key
+            KeyPair keyPair = getkeyPair();
+            jsonWebSignature.setKey(keyPair.getPrivate());
+            jsonWebSignature.setAlgorithmConstraints(new AlgorithmConstraints(PERMIT, deployment.getRequestSignatureAlgorithm()));
+            jsonWebSignature.setAlgorithmHeaderValue(deployment.getRequestSignatureAlgorithm());
+        }
+        jsonWebSignature.sign();
+        return jsonWebSignature;
+    }
+
+    private JsonWebEncryption encryptRequest(JsonWebSignature signedRequest) throws JoseException {
+        JsonWebEncryption jsonEncryption = new JsonWebEncryption();
+        jsonEncryption.setPayload(signedRequest.getCompactSerialization());
+        jsonEncryption.setAlgorithmConstraints(new AlgorithmConstraints(AlgorithmConstraints.ConstraintType.PERMIT, deployment.getRequestEncryptAlgorithm(), deployment.getRequestEncryptEncValue()));
+        jsonEncryption.setAlgorithmHeaderValue(deployment.getRequestEncryptAlgorithm());
+        jsonEncryption.setEncryptionMethodHeaderParameter(deployment.getRequestEncryptEncValue());
+        List<JsonWebKey> jwkList = deployment.getrealmKeySet().getJsonWebKeys();
+        jsonEncryption.setDoKeyValidation(false);
```

Comment:
Why is this needed?

Reply:
Sorry, this is not needed. I have removed it